src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -1205,74 +1212,85 @@ private BoundExpression CreateAndPopulateList(BoundCollectionExpression node, Ty
                 collectionType);
         }
 
-        /// <summary>
-        /// Create and populate a dictionary from a collection expression.
-        /// The collection may or may not have a known length.
-        /// </summary>
-        private BoundExpression CreateAndPopulateDictionary(BoundCollectionExpression node, ImmutableArray<TypeWithAnnotations> typeArguments, ImmutableArray<BoundNode> elements)
+        private BoundExpression VisitDictionaryExpression(BoundCollectionExpression node, NamedTypeSymbol collectionType)
```

Comment:
Consider a more specific name for this (following naming convention as other visit helpers called from `RewriteCollectionExpressionConversion`)
nit: consider asserting the collection type kind